src/anomalib/models/components/base/anomaly_module.py

```diff
@@ -48,10 +52,8 @@ def __init__(self, post_processor: PostProcessor | None = None) -> None:
         self.loss: nn.Module
         self.callbacks: list[Callback]
 
-        self.image_metrics: AnomalibMetricCollection
-        self.pixel_metrics: AnomalibMetricCollection
-
         self.post_processor = post_processor or self.default_post_processor()
+        self.evaluator = self.configure_metrics(metrics)
```

Comment:
How about we have `self.evaluator = self._setup_evaluator(metrics)`
```python
def _setup_evaluator(metrics) -> Evaulator
```
and
`default_metrics->configure_metrics`?

```python
def configure_metrics() -> Sequence[Metrics] | Metrics:
```